frame/header-mmr/src/lib.rs

```diff
@@ -62,83 +62,134 @@ mod tests;
 
 #[frame_support::pallet]
 pub mod pallet {
+	pub mod types {
+		/// The type use for indexing a node
+		pub type NodeIndex = u64;
+	}
+	pub use types::*;
+
 	// --- crates.io ---
 	#[cfg(feature = "std")]
 	use serde::Serialize;
-	// --- github.com ---
-	use mmr::{Error, MMRStore, Merge, Result as MMRResult, MMR};
-	// --- substrate ---
-	use frame_support::pallet_prelude::*;
+	// --- paritytech ---
+	use frame_support::{pallet_prelude::*, weights::Weight};
 	use frame_system::pallet_prelude::*;
-	use sp_runtime::{
-		generic::{DigestItem, OpaqueDigestItemId},
-		traits::{Hash, Header, SaturatedConversion},
-	};
+	use sp_runtime::{generic::DigestItem, traits::SaturatedConversion};
+	#[cfg(test)]
+	use sp_runtime::{generic::OpaqueDigestItemId, traits::Header};
 	use sp_std::prelude::*;
 	// --- darwinia ---
+	use crate::{primitives::*, weights::WeightInfo};
 	use darwinia_header_mmr_rpc_runtime_api::{Proof, RuntimeDispatchInfo};
 	use darwinia_relay_primitives::MMR as MMRT;
 
-	pub const PARENT_MMR_ROOT_LOG_ID: [u8; 4] = *b"MMRR";
+	/// The prefix of [`MerkleMountainRangeRootLog`]
+	pub const LOG_PREFIX: [u8; 4] = *b"MMRR";
 
 	#[pallet::config]
-	pub trait Config: frame_system::Config {}
+	pub trait Config: frame_system::Config {
+		type WeightInfo: WeightInfo;
+
+		/// The offchain-indexing prefix
+		const INDEXING_PREFIX: &'static [u8];
+	}
 
-	/// The MMR size and length of the mmr node list
+	/// Size of the MMR
 	#[pallet::storage]
-	#[pallet::getter(fn mmr_counter)]
-	pub type MMRCounter<T> = StorageValue<_, u64, ValueQuery>;
+	#[pallet::getter(fn mmr_size)]
+	pub type MmrSize<T> = StorageValue<_, NodeIndex, ValueQuery>;
```

Comment:
This changes the metadata and storage definition, will require companion changes in bridger, and need to remove the hack in `substrate-subxt`:

https://github.com/darwinia-network/substrate-subxt/commit/8b5bb8d5f459e4545a2cd665e55b8f1d48dcc276#diff-41a0a2aa11cb8347c60d4b02a2f6b166b0118d9d05e5eb61f46cb67fb0da94b0R186

cc @wuminzhe

Reply:
okay

Reply:
0.3.3.2 depends on wuminzhe/substrate-subxt, so I change it there:
https://github.com/wuminzhe/substrate-subxt/commit/5b76d8dd8b59a4d2292776479bcc0199bb2f3c18

Reply:
maybe not need to change here if use @fewensa 's code. I'll check it.

Reply:
https://github.com/wuminzhe/substrate-subxt/commit/bef909924729c273b24103a2a60a0acf61ea1d4a